client/state/data-layer/wpcom-http/pipeline/remove-duplicate-gets/test/index.js

```diff
@@ -0,0 +1,270 @@
+	} );
+} );
+
+describe( '#addResponder', () => {
+	it( 'should add an `onFailure` action to an empty list', () => {
+		const union = addResponder( {}, { onFailure: failer } );
+
+		expect( union.failures ).to.eql( [ failer ] );
+		expect( union.successes ).to.be.empty;
+	} );
+
+	it( 'should add an `onSuccess` action to an empty list', () => {
+		const union = addResponder( {}, { onSuccess: succeeder } );
+
+		expect( union.failures ).to.be.empty;
+		expect( union.successes ).to.eql( [ succeeder ] );
+	} );
+
+	it( 'should add a "unique" action to an existing list', () => {
+		const union = addResponder( { successes: [ succeeder ] }, { onSuccess: filler } );
+
+		expect( union.successes ).to.eql( [ succeeder, filler ] );
+	} );
+
+	it( 'should merge "duplicate" actions to an existing list', () => {
+		const union = addResponder( { successes: [ succeeder ] }, { onSuccess: succeeder } );
+
+		expect( union.successes ).to.eql( [ succeeder ] );
+	} );
+
+	it( 'should add both `onSuccess` and `onFailure`', () => {
+		const union = addResponder( {
+			failures: [ failer ],
+			successes: [ succeeder ],
+		}, {
+			onFailure: filler,
+			onSuccess: filler,
+		} );
+
+		expect( union.failures ).to.eql( [ failer, filler ] );
+		expect( union.successes ).to.eql( [ succeeder, filler ] );
+	} );
+} );
+
+describe( '#removeDuplicateGets', () => {
+	beforeEach( clearQueue );
+
+	it( 'should pass through non-GET requests', () => {
+		const primed = removeDuplicateGets( { nextRequest: postLike } );
+
+		expect( primed.nextRequest ).to.eql( postLike );
+
+		const processed = removeDuplicateGets( { nextRequest: postLike } );
+
+		expect( processed.nextRequest ).to.eql( postLike );
+	} );
+
+	it( 'should pass through new requests', () => {
+		const processed = removeDuplicateGets( { nextRequest: getSites } );
+
+		expect( processed.nextRequest ).to.eql( getSites );
+	} );
+
+	it( 'should pass through "unique" requests', () => {
+		removeDuplicateGets( { nextRequest: getSites } );
+
+		const processed = removeDuplicateGets( { nextRequest: getPosts } );
+
+		expect( processed.nextRequest ).to.eql( getPosts );
+	} );
+
+	it( 'should drop "duplicate" requests', () => {
+		removeDuplicateGets( { nextRequest: getSites } );
+
+		const processed = removeDuplicateGets( { nextRequest: getSites } );
+
+		expect( processed.nextRequest ).to.be.null;
+	} );
+} );
+
+describe( '#applyDuplicateHandlers', () => {
+	before( clearQueue );
+
+	it( 'should return new requests', () => {
+		removeDuplicateGets( { nextRequest: getSites } );
+
+		const processed = applyDuplicatesHandlers( { originalRequest: getSites } );
+
+		expect( processed.failures ).to.eql( [ getSites.onFailure ] );
+		expect( processed.successes ).to.eql( [ getSites.onSuccess ] );
+	} );
+
+	it( 'should collapse "duplicate" requests having same responders', () => {
+		removeDuplicateGets( { nextRequest: getSites } );
+		removeDuplicateGets( { nextRequest: getSites } );
```

Comment:
this doesn't really reflect how it'll be used. getSites would be different object instances that both came from a `getSites()` action creator. But they won't likely be the same instance.